features/cellular/framework/API/CellularDevice.h

```diff
@@ -70,6 +70,22 @@ class CellularDevice {
      *
      */
     virtual CellularContext *create_context(FileHandle *fh = NULL, const char *apn = NULL) = 0;
+#if (DEVICE_SERIAL && DEVICE_INTERRUPTIN && NSAPI_PPP_AVAILABLE)
```

Comment:
This does need to be conditional on serial+interrupt_in, because it takes UARTSerial, but again doesn't necessarily need to be conditional on PPP. Wouldn't do any harm for an app to call this one, if it knew it was using UARTSerial, but wasn't committed about PPP or not.

Why are the apn and serial parameters reversed? Overload issue?

Reply:
removed #ifdef. _Yes overload when calling internally to avoid copy-paste.